src/snowflake/snowpark/dataframe_transform_functions.py

```diff
@@ -120,3 +221,125 @@ def moving_agg(
                     agg_df = agg_df.with_column(formatted_col_name, agg_col)
 
         return agg_df
+
+    def time_series_agg(
+        self,
+        time_col: str,
+        aggs: Dict[str, List[str]],
+        windows: List[str],
+        group_by: List[str],
+        sliding_interval: str,
+        col_formatter: Callable[[str, str, int], str] = _default_col_formatter,
+    ) -> "snowflake.snowpark.dataframe.DataFrame":
+        """
+        Applies aggregations to the specified columns of the DataFrame over specified time windows,
+        and grouping criteria.
+
+        Args:
+            aggs: A dictionary where keys are column names and values are lists of the desired aggregation functions.
+            windows: Time windows for aggregations using strings such as '7D' for 7 days, where the units are
+                S: Seconds, M: Minutes, H: Hours, D: Days, W: Weeks, T: Months, Y: Years. For future-oriented analysis, use positive numbers,
```

Comment:
Is there a reference for these unit abbreviation? Trying to understand why we use `T` to represent months.

Reply:
Snowflake has used "mm" for months. Instead of accepted 2 characters, we are accepting single character "T"

Reply:
I don't see a good reason that we want to use `T` then. Why don't we follow Snowflake's convention?

Reply:
Discussed offline. Switched it to snowflake convention.